Xamarin.Essentials/WebAuthenticator/WebAuthenticator.android.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Threading.Tasks;
+using Android.Content;
+using Android.Support.CustomTabs;
+
+namespace Xamarin.Essentials
+{
+    public partial class WebAuthenticator
+    {
+        static TaskCompletionSource<AuthResult> tcsResponse = null;
+
+        static Uri uri = null;
+
+        static CustomTabsActivityManager CustomTabsActivityManager { get; set; }
+
+        static Uri RedirectUri { get; set; }
+
+        internal static Task<AuthResult> ResponseTask
+            => tcsResponse?.Task;
+
+        internal static bool OnResume(Intent intent)
+        {
+            // If we aren't waiting on a task, don't handle the url
+            if (tcsResponse?.Task?.IsCompleted ?? true)
+                return false;
+
+            if (intent == null)
+            {
+                tcsResponse.TrySetCanceled();
+                return false;
+            }
+
+            try
+            {
+                var intentUri = new Uri(intent.Data.ToString());
+
+                // Only handle schemes we expect
+                if (!WebUtils.CanHandleCallback(RedirectUri, intentUri))
+                {
+                    tcsResponse.TrySetException(new NullReferenceException("Invalid Redirect URI"));
```

Comment:
Maybe this should be an InvalidOperationException?

Reply:
Agreed.  Changed and made the error message better.